.github/workflows/toolbelt.yaml

```diff
@@ -0,0 +1,157 @@
+# SPDX-License-Identifier: Apache-2.0
+
+name: toolbelt
+
+on:
+  workflow_call: # TODO: Add inputs
+
+permissions:
+  actions: read # for detecting the Github Actions environment.
+  contents: read
+
+jobs:
+  goreleaser:
+    permissions:
+      contents: write # To upload assets to release.
+      packages: write # To publish container images to GHCR
+      id-token: write # needed for signing the images with GitHub OIDC Token
+    runs-on: ubuntu-latest
+    outputs:
+      hashes: ${{ steps.hash.outputs.hashes }}
+      image: ${{ steps.hash.outputs.image }}
+      digest: ${{ steps.hash.outputs.digest }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+        with:
+          fetch-depth: 0
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@e92390c5fb421da1463c202d546fed0ec5c39f20 # v3.1.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up Go
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+        with:
+          go-version: "1.22"
```

Comment:
The Go version may or may not be something else to parameterize.

Reply:
I wonder if this is something we can pull in based on the go.mod or similar.